equinix/resource_fabric_network.go

```diff
@@ -0,0 +1,431 @@
+	mappedChanges := make([]interface{}, len(changes))
+	for _, change := range changes {
+		mappedChange := make(map[string]interface{})
+		mappedChange["href"] = change.Href
+		mappedChange["type"] = string(*change.Type_)
+		mappedChange["uuid"] = change.Uuid
+		mappedChanges = append(mappedChanges, mappedChange)
+	}
+
+	changeSet := schema.NewSet(
+		schema.HashResource(&schema.Resource{Schema: fabricNetworkChangeSch()}),
+		mappedChanges,
+	)
+	return changeSet
+}
+
+func setFabricNetworkMap(d *schema.ResourceData, nt v4.Network) diag.Diagnostics {
+	diags := diag.Diagnostics{}
+	err := equinix_schema.SetMap(d, map[string]interface{}{
+		"name":              nt.Name,
+		"href":              nt.Href,
+		"uuid":              nt.Uuid,
+		"type":              nt.Type_,
+		"scope":             nt.Scope,
+		"state":             nt.State,
+		"operation":         fabricNetworkOperationToTerra(nt.Operation),
+		"change":            simplifiedFabricNetworkChangeToTerra(nt.Change),
+		"location":          equinix_fabric_schema.LocationToTerra(nt.Location),
+		"notifications":     equinix_fabric_schema.NotificationsToTerra(nt.Notifications),
+		"project":           equinix_fabric_schema.ProjectToTerra(nt.Project),
+		"change_log":        equinix_fabric_schema.ChangeLogToTerra(nt.ChangeLog),
+		"connections_count": nt.ConnectionsCount,
+	})
+	if err != nil {
+		return diag.FromErr(equinix_errors.FormatFabricError(err))
+	}
+	return diags
+}
+func getFabricNetworkUpdateRequest(network v4.Network, d *schema.ResourceData) (v4.NetworkChangeOperation, error) {
+	changeOps := v4.NetworkChangeOperation{}
+	existingName := network.Name
+	updateNameVal := d.Get("name")
+
+	log.Printf("existing name %s, Update Name Request %s ", existingName, updateNameVal)
+
+	if existingName != updateNameVal {
+		changeOps = v4.NetworkChangeOperation{Op: "replace", Path: "/name", Value: &updateNameVal}
+	} else {
+		return changeOps, fmt.Errorf("nothing to update for the Fabric Network: %s", existingName)
+	}
+	return changeOps, nil
+}
+func resourceFabricNetworkUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*config.Config).FabricClient
+	ctx = context.WithValue(ctx, v4.ContextAccessToken, meta.(*config.Config).FabricAuthToken)
+	dbConn, err := waitUntilFabricNetworkIsProvisioned(d.Id(), meta, ctx)
+	if err != nil {
+		if !strings.Contains(err.Error(), "500") {
+			d.SetId("")
+		}
+		return diag.Errorf("either timed out or errored out while fetching Fabric Network for uuid %s and error %v", d.Id(), err)
+	}
+	// TO-DO
+	update, err := getFabricNetworkUpdateRequest(dbConn, d)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	updates := []v4.NetworkChangeOperation{update}
+	_, res, err := client.NetworksApi.UpdateNetworkByUuid(ctx, updates, d.Id())
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("error response for the Fabric Network update, response %v, error %v", res, equinix_errors.FormatFabricError(err)))
```

Comment:
We can just put the `equinix_errors.FormatFabricError(err)` in the diag.FromErr. It will provide enough context.

Reply:
This one line should be: 
`diag.FromErr(equinix_errors.FormatFabricError(err))`

1. If the err is directly from Fabric-Go, then use the `equinix_errors.FormatFabricError(err)` on it. Otherwise leave it as just err.
2. You can provide context if the error is from the waitMethods, otherwise the formatting of the err is enough context.
3. If you do need to add context, use `diag.Errorf(...)` not `diag.FromErr(fmt.Errorf(...))`

Reply:
This line was never updated.